hivelocity/instances_v2_test.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hivelocity
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	hv "github.com/hivelocity/hivelocity-client-go/client"
+	"github.com/hivelocity/hivelocity-cloud-controller-manager/client"
+	"github.com/hivelocity/hivelocity-cloud-controller-manager/client/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	cloudprovider "k8s.io/cloud-provider"
+)
+
+func Test_GetHivelocityDeviceIDFromNode(t *testing.T) {
+	var tests = []struct {
+		providerID    string
+		wantDeviceID  int32
+		wantErrString string
+	}{
+		{
+			providerID:    "",
+			wantDeviceID:  0,
+			wantErrString: "missing prefix \"hivelocity://\" in node.Spec.ProviderID \"\"",
+		},
+		{
+			providerID:    "hivelocity://12345",
+			wantDeviceID:  12345,
+			wantErrString: "",
+		},
+	}
+	var node = &corev1.Node{}
```

Comment:
At the moment it looks like this:

```
	node := newNode()
	for _, tt := range tests {
		node.Spec.ProviderID = tt.providerID
```

Yes, I would like to keep it like this. 

But I don't have a strong opinion here.

I can change it, if you want me to change it.

Adding a new parameter to newNode(), then should what should it look like?

v1: int, like 1234
v2: string, like "1234"
v3: string including providerName: like "hivelocity://1234".

I could write a sub-function, which is only available in the method, so that other places don't need to get updated.

Reply:
solved in 08d88a7